addons/web/static/tests/views/graph_tests.js

```diff
@@ -979,6 +979,150 @@ QUnit.module('Views', {
         graph.destroy();
     });
 
+    QUnit.test('clicking on bar and pie charts triggers a do_action', async function (assert) {
+        assert.expect(5);
+
+        const graph = await createView({
+            View: GraphView,
+            model: "foo",
+            data: this.data,
+            arch: '<graph><field name="bar"/></graph>',
+            intercepts: {
+                do_action: function (ev) {
+                    assert.deepEqual(ev.data.action, {
+                        context: {},
+                        domain: [["bar", "=", true]],
+                        name: true, // ?
+                        res_model: "foo",
+                        target: 'current',
+                        type: 'ir.actions.act_window',
+                        view_mode: 'list',
+                        views: [[false, 'list'], [false, 'form']],
+                    }, "should trigger do_action with correct action parameter");
+                }
+            },
+        });
+        await testUtils.nextTick(); // wait for the graph to be rendered
+
+        // bar mode
+        assert.strictEqual(graph.renderer.state.mode, "bar", "should be in bar chart mode");
+        assert.checkDatasets(graph, ['domain'], {
+            domain: [[["bar", "=", true]], [["bar", "=", false]]],
+        });
+
+        let myChart = graph.renderer.chart;
+        let meta = myChart.getDatasetMeta(0);
+        let rectangle = myChart.canvas.getBoundingClientRect();
+        let point = meta.data[0].getCenterPoint();
+        let mouseMoveEvent = new MouseEvent('click', {
```

Comment:
Better to use existing helpers: `testUtils.dom.triggerEvent` with `MouseEvent` typed events allows to provide an `eventAttrs` dict as third argument which can set `clientX` and `clientY` keys to the final triggered event object

Reply:
And the event here is called `mouseMoveEvent` while it triggers a click...